tests/network/libs/vm.py

```
@@ -0,0 +1,29 @@
+    Args:
+        vms: VMs to start.
+
+    Returns:
+        The same tuple of VMs, all running with guest agent connected.
+    """
+    for vm in vms:
+        try:
+            vm.start()  # type: ignore[no-untyped-call]
+        except ApiException as vm_exception:
+            if "VM is already running" in vm_exception.body:
+                LOGGER.warning(f"VM {vm.name} is already running")
```

Ruff (0.15.13)

[warning] Logging statement uses f-string

(G004)